management/management_request_test.go

```diff
@@ -0,0 +1,91 @@
+package management
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewRequest(t *testing.T) {
+	testCases := []struct {
+		name          string
+		method        string
+		uri           string
+		payload       interface{}
+		options       []RequestOption
+		expectedError error
+	}{
+		{
+			name:          "Valid GET request",
+			method:        "GET",
+			uri:           "https://example.com",
+			payload:       nil,
+			options:       nil,
+			expectedError: nil,
+		},
+		{
+			name:          "Valid POST request with payload",
+			method:        "POST",
+			uri:           "https://example.com",
+			payload:       map[string]string{"key": "value"},
+			options:       nil,
+			expectedError: nil,
+		},
+		{
+			name:          "Invalid payload encoding",
+			method:        "POST",
+			uri:           "https://example.com",
+			payload:       make(chan int), // Unsupported type
+			options:       nil,
+			expectedError: fmt.Errorf("encoding request payload failed"),
+		},
+		{
+			name:          "Valid DELETE request",
+			method:        "DELETE",
+			uri:           "https://example.com",
+			payload:       nil,
+			options:       nil,
+			expectedError: nil,
+		},
+		{
+			name:          "Valid PUT request with payload",
+			method:        "PUT",
+			uri:           "https://example.com",
+			payload:       map[string]string{"key": "value"},
+			options:       nil,
+			expectedError: nil,
+		},
+		{
+			name:          "Valid PATCH request with payload",
+			method:        "PATCH",
+			uri:           "https://example.com",
+			payload:       map[string]string{"key": "value"},
+			options:       nil,
+			expectedError: nil,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			m := &Management{}
+			request, err := m.NewRequest(context.Background(), testCase.method, testCase.uri, testCase.payload, testCase.options...)
+			if testCase.expectedError != nil {
+				assert.Error(t, err)
+				assert.Contains(t, err.Error(), testCase.expectedError.Error())
```

Comment:
Any chance we can make the `expectedError` a string and then use `assert.EqualError` instead? This would reduce the assertions to just 1 line.